algorithms/pact/pact_ops.py

```diff
@@ -229,20 +229,24 @@ class ChannelwiseThreshold(nn.Module):
     class MyChannelwiseThreshold(torch.autograd.Function):
 
         @staticmethod
-        def forward(ctx, x, thresh_lo, thresh_hi):
+        def forward(ctx, x, thresh_lo, thresh_hi, signed_out):
             tmp1 = -1*(x < thresh_lo).type_as(x)
             tmp2 = (x >= thresh_hi).type_as(x)
 
             return tmp1 + tmp2
 
         @staticmethod
-        @parse_args('v', 't', 't')
-        def symbolic(g, x, thresh_lo, thresh_hi):
-            thresh_lo_ = g.op("Constant", value_t=thresh_lo)
-            thresh_hi_ = g.op("Constant", value_t=thresh_hi)
-            return g.op("PACTOps::ChannelwiseThreshold2d", x, thresh_lo_t=thresh_lo, thresh_hi_t=thresh_hi)
-
-    def __init__(self, thresh_lo : torch.Tensor, thresh_hi : torch.Tensor, n_dim : Literal[1,2] = 2):
+        @parse_args('v', 'is', 'is', 'i')
+        def symbolic(g, x, thresh_lo, thresh_hi, signed_out):
+            return g.op("PACTOps::ChannelwiseThreshold2d",
+                        x, thresh_lo_i=thresh_lo, thresh_hi_i=thresh_hi,
+                        signed_out_i=signed_out).setType(x.type().with_sizes(_get_tensor_sizes(x)))
```

Comment:
Does this set the output's shape to the input's shape? Is this required?

Reply:
It does - I _believe_ it's needed to correctly propagate the shape, but that was not working prior to PyTorch 2.0 in general; now it does but I'm going to verify again that this is actually required.

Reply:
Checked - it is needed for correct shape propagation.